crates/ruff_db/src/system/path.rs

```diff
@@ -564,6 +564,106 @@ impl ruff_cache::CacheKey for SystemPathBuf {
     }
 }
 
+/// A slice of a virtual path on [`System`](super::System) (akin to [`str`]).
+#[repr(transparent)]
+pub struct SystemVirtualPath(str);
+
+impl SystemVirtualPath {
+    pub fn new(path: &str) -> &SystemVirtualPath {
+        // SAFETY: SystemVirtualPath is marked as #[repr(transparent)] so the conversion from a
+        // *const str to a *const SystemVirtualPath is valid.
+        unsafe { &*(path as *const str as *const SystemVirtualPath) }
+    }
+
+    /// Converts the path to an owned [`SystemVirtualPathBuf`].
+    pub fn to_path_buf(&self) -> SystemVirtualPathBuf {
+        SystemVirtualPathBuf(self.0.to_string())
+    }
+
+    /// Returns the std path for the file.
+    #[inline]
+    pub fn as_std_path(&self) -> &Path {
+        Path::new(self.as_str())
+    }
```

Comment:
I don't think we can safely offer this method. For example, calling it for `untitled://name` would fail.

Reply:
Is it because the file doesn't really exists and that the methods on `std::path::Path` would error?

Reply:
I don't think it would error. It's more that operations like `components` may not work as intended.